libp2p/wire.nim

```diff
@@ -20,21 +20,45 @@ else:
 
 const
   TRANSPMA* = mapOr(
-    mapAnd(IP, mapEq("udp")),
-    mapAnd(IP, mapEq("tcp")),
-    mapAnd(mapEq("unix"))
+    UDP,
+    TCP,
+    UNIX,
   )
 
   RTRANSPMA* = mapOr(
-    mapAnd(IP, mapEq("tcp")),
-    mapAnd(mapEq("unix"))
+    TCP,
+    UNIX
   )
 
+proc resolveDns(ma: MultiAddress, domain: Domain = Domain.AF_UNSPEC, prefix: string = ""): MaResult[TransportAddress]
+  {.raises: [Defect, ResultError[string]]} =
+  try:
+    var
+      dnsbuf: array[256, byte]
+      dnsval: string
+      pbuf: array[2, byte]
+    if ma[0].tryGet().protoArgument(dnsbuf).tryGet() == 0:
+      err("Invalid DNS argument")
+    else:
+      dnsval.setLen(dnsbuf.len)
+      copyMem(addr dnsval[0], addr dnsbuf[0], dnsbuf.len)
+      if ma[1].tryGet().protoArgument(pbuf).tryGet() == 0:
+        err("Incorrect port number")
+      else:
+        let port = Port(fromBytesBE(uint16, pbuf))
+        let resolvedAddresses = resolveTAddress(prefix & dnsval, port, domain)
+        if resolvedAddresses.len > 0:
+          ok(resolvedAddresses[0])
```

Comment:
Think this should return the full set of resolved addresses.

Reply:
Yup, the issue with using multiple resolved addresses is that the resolving must happen before-hand (eg, in the dialer -> internalConnect), so that the dialer can try out the different addresses.
I'll do that then